### ISSUE TITLE
failed to compile simple example

### ISSUE THREAD

**Issue by dan-42** [nxxm/nxxm#10]
Hey, i see the following issue:

```
C:\Users\friedrich\git\tmp\nxxm_play>nxxm .
shipping cmake.
- shipping cmake done.
shipping platform libs recipes.
- shipping platform libs recipes done.
shipping make.
- shipping make done.
shipping toolchain configs.
- shipping toolchain configs done.
shipping python.
- shipping python done.
shipping nodejs.
- shipping nodejs done.
shipping emsdk.
- shipping emsdk done.
initializing emsdk.
- done.
- done.
- done.
scanning project...
- done.
convention build summary :
==========================

  nxxm_play library
  =================

   no .cpp files, header only.

  executables
  ===========

    * ./main
       + cpps : main.cpp


e . C:/Users/friedrich/git/tmp/nxxm_play/main.cpp
ÔùÅ [ok] CMake build recipes inferred by convention in "build\recipe"
Compiling "." for wasm-cxx17
- ÔùÅ [err] configuring.
-- Configuring incomplete, errors occurred!
Failed to create ConsoleBuf!
GetStdHandle(STD_INPUT_HANDLE) returned INVALID_HANDLE_VALUE
Failed to create ConsoleBuf!
GetStdHandle(STD_INPUT_HANDLE) returned INVALID_HANDLE_VALUE
CMake Error at C:/.nxxm/0000000/polly/wasm.cmake:103 (message):
  Failed to fetch compiler version information with command
  "'C:/.nxxm/0000000/emsdk/emscripten/1.38.11/emcc.bat' -v"! Process returned
  with error code The system cannot find the file specified.
Call Stack (most recent call first):
  C:/.nxxm/0000000/polly/wasm-cxx17.cmake:19 (include)
  C:/Users/friedrich/git/tmp/nxxm_play/build/wasm-cxx17/sysroot/proxy.cmake:2 (include)
  C:/.nxxm/0000000/cmake/share/cmake-3.12/Modules/CMakeDetermineSystem.cmake:94 (include)
  CMakeLists.txt:2 (project)



[hunter ** INTERNAL **] Configure project failed.
[hunter ** INTERNAL **] To reproduce the error run: C:/.nxxm/0000000/cmake/bin/cmake.exe -HC:/.hunter/_Base/Download/Hunter/0.0.3/2d89a86 -BC:/.hunter/_Base/Download/Hunter/0.0.3/2d89a86/Build -GMinGW Makefiles -DCMAKE_TOOLCHAIN_FILE=C:/Users/friedrich/git/tmp/nxxm_play/build/wasm-cxx17/sysroot/proxy.cmake
[hunter ** INTERNAL **] In directory C:/.hunter/_Base/Download/Hunter/0.0.3/2d89a86
[hunter ** INTERNAL **] [Directory:C:/Users/friedrich/git/tmp/nxxm_play/build/recipe]

------------------------------ WIKI -------------------------------
    https://github.com/ruslo/hunter/wiki/error.internal
-------------------------------------------------------------------

CMake Error at C:/.nxxm/0000000/cmake/share/cmake-3.12/Modules/HunterGate.cmake:87 (message):
Call Stack (most recent call first):
  C:/.nxxm/0000000/cmake/share/cmake-3.12/Modules/HunterGate.cmake:97 (hunter_gate_wiki)
  C:/.nxxm/0000000/cmake/share/cmake-3.12/Modules/HunterGate.cmake:326 (hunter_gate_internal_error)
  C:/.nxxm/0000000/cmake/share/cmake-3.12/Modules/HunterGate.cmake:510 (hunter_gate_download)
  CMakeLists.txt:10 (HunterGate)


ÔùÅ [err] Dynamic exception type: class std::runtime_error
std::exception::what: configure failed in C:/Users/friedrich/git/tmp/nxxm_play/build/wasm-cxx17/bin
```

folder contains single ``` main.cpp ``` with:
```
#include <iostream>

int main(int argc, char* argv[])
{
  std::cout << "hello world" << std::endl;
  return 0;
}
```

system is Windows 10
 installed is 
 * CMake 3.16 and is on the system path
 * Ninja and is on the system path
 * Visual Studio  2015, 2017 and 2019

If anything else is needed, let me know

**Comment by daminetreg** [comment]
Hi @dan-42 thanks for playing with nxxm.

Can you tell me if there is a ‘c:/.nxxm/0000000/emsdk/emscripten/1.38.11/emcc.bat’ file on your machine ?

We have the same test environment on the CI. It should work and doesn’t require anything installed on your machine.

**Comment by AnotherFoxGuy** [comment]
I've run into the same problem, the issue seems to be that the emsdk URLs have changed

```
nxxm . --verbose
initializing emsdk.
- {..          .}Error downloading URL 'https://s3.amazonaws.com/mozilla-games/emscripten/packages/emsdk_windows_update.zip': HTTP Error 403: Forbidden
- {...          }Fetching all tags from Emscripten Github repository...
- {..          .}Done. 180 tagged releases available, latest is 1.39.3.
Fetching all tags from Binaryen Github repository...
- {        ...  }Done. 90 tagged Binaryen releases available, latest is 1.39.1.
Fetching all precompiled tagged releases..
- {...          }Error downloading URL 'https://s3.amazonaws.com/mozilla-games/emscripten/packages/llvm/tag/win_32bit/index.txt': HTTP Error 403: Forbidden
- {       ...   }Error downloading URL 'https://s3.amazonaws.com/mozilla-games/emscripten/packages/llvm/tag/win_64bit/index.txt': HTTP Error 403: Forbidden
- {        ...  }Downloading: C:/.nxxm/0000000/emsdk/upstream/lkgr.json from https://storage.googleapis.com/wasm-llvm/builds/linux/lkgr.json, 3657 Bytes
- done.          00%]
- {   ...       }Fetching all tags from Emscripten Github repository...
Done. 180 tagged releases available, latest is 1.39.3.
- Fetching all tags from Binaryen Github repository...
- { ...         }Done. 90 tagged Binaryen releases available, latest is 1.39.1.
Fetching all precompiled tagged releases..
- {       ...   }Error downloading URL 'https://s3.amazonaws.com/mozilla-games/emscripten/packages/llvm/tag/win_32bit/index.txt': HTTP Error 403: Forbidden
- { ...         }Error downloading URL 'https://s3.amazonaws.com/mozilla-games/emscripten/packages/llvm/tag/win_64bit/index.txt': HTTP Error 403: Forbidden
- {   ...       }Downloading: C:/.nxxm/0000000/emsdk/upstream/lkgr.json from https://storage.googleapis.com/wasm-llvm/builds/linux/lkgr.json, 3657 Bytes
- {     ...      Error: No tool or SDK found by name 'sdk-1.38.11-64bit'.
- done.
- {   ...       }Writing .emscripten configuration file to user home directory C:\Users\Edgar/
- {     ...     }Error: No tool or SDK found by name 'sdk-1.38.11-64bit'.
- done.
Checking if "C:/.nxxm\0000000\emsdk\emscripten\1.38.11\emcc.py" is up-to-date.
Downloading https://raw.githubusercontent.com/nxxm/emscripten/nxxm/1.38.11/emcc.py as "C:/.nxxm\0000000\emsdk\emscripten\1.38.11\emcc.py"
```

**Comment by daminetreg** [comment]
Thank you very much. I was aware of the change but recent CI jobs didn’t fail. Seems to be an issue with some caching.

I will release an hotfix ASAP. I’ll keep you updated here. Thank you very much for the great trace.

**Comment by dan-42** [comment]
> Can you tell me if there is a ‘c:/.nxxm/0000000/emsdk/emscripten/1.38.11/emcc.bat’ file on your machine ?

sry for the late reply, must have missed the notification.
Yes i do have a ‘c:/.nxxm/0000000/emsdk/emscripten/1.38.11/emcc.bat’ on my system

**Comment by daminetreg** [comment]
The newer version solves this problem, windows version doesn't work well yet (we are working on it) with Boost for the moment but linux and macOS have full coverage, the alpha version solving this issue is here : https://github.com/nxxm/nxxm/releases/tag/v0.0.8

**Comment by dan-42** [comment]
@daminetreg thanks for the update on that issue 👍

**Comment by daminetreg** [comment]
This is fixed in stable version v0.0.10. Re-running nxxm will automatically propose to reupdate it, or you can get the latest version for windows by dropping this into a powershell : 

```powershell
. { iwr -useb https://raw.githubusercontent.com/nxxm/nxxm/master/install/install_for_windows.ps1 } | iex
```

On linux / macOS : 
```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/nxxm/nxxm/master/install/install_for_macos_linux.sh)"
```
